libs/shared/flagd-core/src/lib/flagd-core.ts

```diff
@@ -91,35 +92,47 @@ export class FlagdCore {
       }
     }
 
-    const defaultVariant = flag.defaultVariant;
-
-    let resolvedVariant;
+    let variant;
     let reason;
 
-    if (flag.targetingString == undefined) {
-      resolvedVariant = flag.variants.get(defaultVariant);
+    if (!flag.targeting) {
+      variant = flag.defaultVariant;
       reason = StandardResolutionReasons.STATIC;
     } else {
-      // todo - targeting evaluation
-      // till targeting is handled, return the static result
-      resolvedVariant = flag.variants.get(defaultVariant);
-      reason = StandardResolutionReasons.STATIC;
+      let targetingResolution;
+      try {
+        targetingResolution = this._targeting.applyTargeting(flagKey, flag.targeting, evalCtx);
+      } catch (e) {
+        console.error(`Error evaluating targeting rule for flag ${flagKey}, falling back to default`, e);
+        targetingResolution = null;
+      }
+
+      if (!targetingResolution) {
+        variant = flag.defaultVariant;
+        reason = StandardResolutionReasons.DEFAULT;
+      } else {
+        variant = targetingResolution;
+        reason = StandardResolutionReasons.TARGETING_MATCH;
```

Comment:
Do we have any way of detecting if the reason is SPLIT?

Reply:
we don't detect this even at flagd. And I don't see a clear cut path to enable this 🤔

Reply:
Yeah, that's what I thought. Thanks.